sonar-plugin-api/src/main/java/org/sonar/api/batch/rule/Checks.java

```diff
@@ -166,8 +170,7 @@ private static void configureFields(ActiveRule activeRule, Object check) {
     for (Map.Entry<String, String> param : activeRule.params().entrySet()) {
       Field field = getField(check, param.getKey());
       if (field == null) {
-        throw new IllegalStateException(
-          String.format("The field '%s' does not exist or is not annotated with @RuleProperty in the class %s", param.getKey(), check.getClass().getName()));
+        LOG.error("The field '{}' does not exist or is not annotated with @RuleProperty in the class {}", param.getKey(), check.getClass().getName());
```

Comment:
I think it'd be good to add a comment explaining why we don't fail.

Reply:
nice to have: we could check if we are running in SC by passing a `SonarRuntime` to `CheckFactory`, and `SonarProduct` to this class, and then log an error if running on SC.

Reply:
It seems like `SonarProduct` currently only contains `SonarQube` and `SonarLint` as possible values.

Reply:
While we are at doing no so great code, we might as well test for SonarProduction=SonarQube and version=8.0, which will only match SonarCloud...

Reply:
FYI The "proper" way to detect SonarCloud at runtime is to look at SonarEdition... maybe this is a good time to fix that, deprecated `SonarEdition.SONARCLOUD` and add `SonarProduct.SONARCLOUD`